### PR TITLE
Remove aws iam authenticator

### PR DESCRIPTION
**Pull request by Skarlso**
### Description

Another attempt at fixing https://github.com/weaveworks/eksctl/issues/5140.

New test run with the updated image and this branch pushed to Origin: https://github.com/weaveworks/eksctl-ci/actions/runs/2486363379

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: build/docker/Dockerfile [hunk 1]
```diff
@@ -41,15 +41,16 @@ ENV GITHUB_CLI_VERSION 1.14.0
 RUN curl --silent --location "https://github.com/cli/cli/releases/download/v${GITHUB_CLI_VERSION}/gh_${GITHUB_CLI_VERSION}_linux_amd64.tar.gz" \
     | tar xvz -C /out/usr/local/bin --strip-components=2 gh_${GITHUB_CLI_VERSION}_linux_amd64/bin/gh && gh --version
 
-# Remaining dependencies are controlled by go.mod
 WORKDIR /src
-ENV CGO_ENABLED=0 GOPROXY=https://proxy.golang.org,direct
 
-# The authenticator is a runtime dependency, so it needs to be in /out
-ENV AWS_IAM_AUTH_VERSION 0.5.2
+# The authenticator is a runtime dependency.
```
Comment (Skarlso) [inline build/docker/Dockerfile:46]: I had to re-order this to trigger a completely new build and a new image id. I definitely want to redo the whole building image thing by adding a version tag which will trigger a new build of a new layer.

**Comment by Skarlso** [comment]
Test run of the release with this container: https://github.com/weaveworks/eksctl/actions/runs/2493667192

File: .github/actions/setup-identity/action.yaml [hunk 2]
```diff
@@ -16,9 +16,11 @@ outputs: {}
 runs:
   using: "composite"
   steps:
+    # add safe.directory is repeated here because it doesn't propagate over the checkout action
     - name: Setup git config
       shell: bash
       run: |
+        git config --global --add safe.directory '/__w/eksctl/eksctl'
```
Comment (Skarlso) [inline .github/actions/setup-identity/action.yaml:23]: Without this, the release breaks. Turns out, it doesn't work ( no idea why it did during testing but now it doesn't lol ).

It's not propagating the setting. And `GITHUB_WORKSPACE` is not this folder so setting that is actually not the thing that fixed the thing apparently.

This is however so fingers crossed that nothing else will break here. I ran all action short on doing an actual release, so I'm hoping there won't be any more problems.

There are ample of issues open that concern this issue on checkout's GitHub page which I linked to in our issue, so tracking that it is sufficient. This is the suggested workaround.

File: build/docker/Dockerfile (hunk 1 above)
Comment (cPu1) [reply build/docker/Dockerfile:46]: Changing `ENV AWS_IAM_AUTH_VERSION 0.5.7` alone would result in the layers below it to be rebuilt. Reordering the instructions shouldn't be required.

File: .github/workflows/release-candidate.yaml [hunk 3]
```diff
@@ -7,14 +7,13 @@ jobs:
...
     runs-on: ubuntu-latest
-    container: weaveworks/eksctl-build:0eb3a29941b9740f055ecb3fd9b663762a7ee9bd
+    container: weaveworks/eksctl-build:1298c3979281b6438cfce89f3215cc0001e6dfc3
     steps:
       - name: Checkout
         uses: actions/checkout@v3
         with:
           token: ${{ secrets.WEAVEWORKSBOT_TOKEN }}
           fetch-depth: 0
-          set-safe-directory: '$GITHUB_WORKSPACE'
```
Comment (Himangini) [inline .github/workflows/release-candidate.yaml:17]: why are we removing these which were added required here https://github.com/weaveworks/eksctl/pull/5424/files

Comment (Skarlso) [reply .github/workflows/release-candidate.yaml:17]: Please see the comment I left here https://github.com/weaveworks/eksctl/pull/5402#discussion_r899230635

File: build/docker/Dockerfile (hunk 1 above)
Comment (Skarlso) [reply build/docker/Dockerfile:46]: It was the layers above this.

Comment (Skarlso) [reply build/docker/Dockerfile]: Removed. :)

**Review by cPu1 — APPROVED** [review]

**Review by Himangini — APPROVED** [review]
👍🏻 Lets get this out , hopefully it will work this time 🤞🏻

**Pull request merged**